app/models/comment.rb

```diff
@@ -47,13 +47,16 @@ def hidden(user)
     end
 
     def visible(user)
-      if user.is_moderator?
-        where("comments.score >= ? or comments.is_sticky = true", user.comment_threshold)
-      elsif user.is_janitor?
-        where("(comments.score >= ? or comments.is_sticky = true) and (comments.is_sticky = true or comments.is_hidden = false or comments.creator_id = ?)", user.comment_threshold, user.id)
-      else
-        where("(comments.score >= ? or comments.is_sticky = true) and (comments.is_hidden = false or comments.creator_id = ?)", user.comment_threshold, user.id)
+      q = where("comments.score >= ? or comments.is_sticky = true", user.comment_threshold)
+      unless user.is_moderator?
+        q = q.joins(:post).where("comments.is_sticky = true or posts.is_comment_disabled = false")
```

Comment:
I would argue that the users should still see their own comments even if the post has the comment section disabled.
As it is, this may create a strange situation where the user may have the profile page may say that they have created comments, but clicking on the comment count may show no results.

Not showing users their own comments on the post page is fine with me, though.